tensorboard/plugins/profile/input_pipeline_analyzer/input-pipeline-analyzer.html

```diff
@@ -0,0 +1,373 @@
+        computed: '_getToggleButtonText(_show_host_side_table)'
+      },
+    },
+    _summary_text: String,
+    _infeed_percent_average: String,
+    _infeed_percent_stddev: String,
+    _infeed_percent_minimum: String,
+    _infeed_percent_maximum: String,
+    _steptime_ms_average: String,
+    _steptime_ms_stddev: String,
+    _steptime_ms_minimum: String,
+    _steptime_ms_maximum: String,
+    _host_conclusion: String,
+    onClick: function(e) {
+      this.set('_show_host_side_table', !this._show_host_side_table);
+    },
+    _reloadToolData: function(run) {
+      this._requestManager.request(addParams(
+          getRouter().pluginRoute('profile', '/data'), {tag: 'input_pipeline_analyzer', run})
+      ).then((data) => {
+        if (data) {
+          this.set('_data', data);
+        }
+      });
+    },
+    _getToggleButtonText: function(show_host_side_table) {
+      return (show_host_side_table ? 'Hide' : 'Show') + ' Input Op Statistics';
+    },
+
+    /* Update view according to new data */
+    _updateView: function() {
+      var deviceJson = this._data[0];
+      var hostJson = this._data[1];
+      this.set('_summary_text', deviceJson.p['summary_text']);
+      this.set('_infeed_percent_average', deviceJson.p['infeed_percent_average']);
+      this.set('_infeed_percent_stddev', deviceJson.p['infeed_percent_standard_deviation']);
+      this.set('_infeed_percent_minimum', deviceJson.p['infeed_percent_minimum']);
+      this.set('_infeed_percent_maximum', deviceJson.p['infeed_percent_maximum']);
+      this.set('_steptime_ms_average', deviceJson.p['steptime_ms_average']);
+      this.set('_steptime_ms_stddev', deviceJson.p['steptime_ms_standard_deviation']);
+      this.set('_steptime_ms_minimum', deviceJson.p['steptime_ms_minimum']);
+      this.set('_steptime_ms_maximum', deviceJson.p['steptime_ms_maximum']);
+      this.set('_host_conclusion', hostJson.p['conclusion']);
+      this.customStyle['--summary-color'] =  deviceJson.p['summary_color'];
+      this.updateStyles();
+      this._showDeviceStepChart(deviceJson);
+      this._showDeviceInfeedChart(deviceJson);
+      this._showHostChart(hostJson);
+      this._showHostTable(hostJson);
+    },
+
+    /* Plots a graph of step time (in ms) against step number */
+    _showDeviceStepChart : function(deviceJson) {
+      var deviceStepSeries = [];
+      var hostStepSeries = [];
+      var maxStepTime = 0;
+      deviceJson.rows.forEach(function(row, index) {
+        var step = Number(row.c[0].v);
+        deviceStepSeries.push({"scalar": row.c[1].v,
+                               "step": step,
+                               "low_watermark": 0})
+        hostStepSeries.push({"scalar": row.c[1].v + row.c[2].v,
+                             "step": step,
+                             "low_watermark": row.c[1].v})
+        maxStepTime = Math.max(maxStepTime, row.c[1].v + row.c[2].v)
+      })
+      var chart = document.querySelector('#device_step_chart');
```

Comment:
optional: Because the ID name contains no hyphens, you can actually use this.$.device_step_chart

Reply:
done.